deep_qa/models/sentence_selection/siamese_sentence_selector.py

```diff
@@ -56,18 +65,36 @@ def _build_model(self):
         # shape: (batch size, num_sentences, num_sentence_words, embedding size)
         sentences_embedding = self._embed_input(sentences_input)
 
-        # We encode the question embeddings with some encoder.
+        # We encode the question embedding with some more seq2seq layers
+        modeled_question = question_embedding
+        for i in range(self.num_seq2seq_layers):
+            hidden_layer = self._get_seq2seq_encoder(name="question_seq2seq_{}".format(i),
```

Comment:
do you think it's preferable to share weights / use the same encoder, or keep them separate? I agree that passing a flag that indicates whether to share weights would work, but it's hard to make it very fine-grained without making the config overly verbose / cumbersome